### ISSUE TITLE
System bypases BIOS VRAM setup => Sluggish and laggy desktop

### ISSUE THREAD

**Issue by roired** [helloSystem/hello#277]
Good day,

This is a Ryzen 5700G box running HelloSystem 0.7.0 off an external SSD USB with 32 GB of RAM. Assigend 2 GB to VRAM in BIOS.

The system behaves quite poorly, anything that is drawn on the screen takes even seconds to appear. 
- Even typing letters have to wait for a second from hitting the key to the actual letter to appear on screen. 
- Mouse cursor jumps from location to location making it hard to click on any menu/icon

Querying the About System reports the system has 32GB of RAM, instead of 30 GB, leaving the 2 GB of RAM used theoretically by the iGPU. I presume HelloSystem is bypassing BIOS settings and finds no memory assigned to the GPU thus its performance is a PITA?

I would appreciate any help on how to solve this issue, and let me know if I can provide more info to identify why this happens.

Regards,
RR

**Comment by probonopd** [comment]
Hello @roired, thanks for reporting this.

Please run:

```
sysctl dev.vgapci | grep pnpinfo
```

and post the result here.
Thanks!

**Comment by probonopd** [comment]
And please upload a Hardware Proble using 0G155 or later (this has the Hardware Probe URL bug fixed) and share the URL. Thanks!

**Comment by ghost** [comment]
Good day @probonopd ,

here is the outcome of that terminal command:
`dev.vgapci.0.%pnpinfo: vendor=0x1002 device=0x1638 subvendor=0x1458 subdevice=0xd000 class=0x030000
`

by 0G155 you mean a new release of HelloSystem?, I'll have to download a new ISO and reinstall the whole thing?

By the way, maybe related, or maybe the cause, not sure, but in order to boot the USB with HelloSystem I need to manually select it in the BIOS boot menu, it does not boot automatically as the other USB SSDs with other oses [Kinoite, Haiku, Windows]. Maybe something in the boot process is not behaving as expected?


Regards,
RR

**Comment by probonopd** [comment]
> by 0G155 you mean a new release of HelloSystem?, I'll have to download a new ISO and reinstall the whole thing?

Yes, I mean the latest pre-release build in the 0G... series that will become 0.7.0 in due time. You do not need to install it in order to submit a Hardware Probe, just write it to a USB stick and boot from it, e.g., using the Create Live Media utility.

**Comment by probonopd** [comment]
> `dev.vgapci.0.%pnpinfo: vendor=0x1002 device=0x1638 subvendor=0x1458 subdevice=0xd000 class=0x030000`

So this is the GPU in question:

https://bsd-hardware.info/?d=helloSystem&id=pci:1002-1638-1458-d000

This AMD chipset is called "Renoir/Cezanne".

Apparently hardware support in FreeBSD is possible with version 5.5 of `drm-kmod` but it looks like we are still on 5.4:
https://hauweele.net/~gawen/blog/?p=2607

Maybe @gawen947 can send us a pkg for version 5.5 of `drm-kmod` for FreeBSD 13.0-RELEASE? 

@evadot may know whether the symptoms you are describing would be addressed by 5.5.

**Comment by ghost** [comment]
So it looks like the Hardware Probe finally got uploaded?

Then I presume I'll have to wait to see the desktop behave :]

Thanks @probonopd ,
Regards,
RR

**Comment by probonopd** [comment]
Also,

```
Loading /usr/local/lib/xorg/modules/drivers/vesa_drv.so
```

gets loaded by Xorg rather than the Radeon module. So possibly the PCI IDs of Renoir/Cezanne devices need to be added to https://github.com/nomadbsd/NomadBSD/blob/master/config/etc/initgfx_device.db as well once Renoir/Cezanne support has landed in `drm-kmod`.

I suppose that adding the ID before we get the new `drm-kmod` will not improve things, but if you would like to try it out, it might be worth a try. Add your Device ID `0x1638` to `radeon_ids()` to `/etc/initgfx_device.db`, delete `/var/initgfx_config.id`, and reboot. Worst thing that could happen is that Xorg won't start anymore, so better enable SSH access to your machine beforehand should you need it to revert the changes.

cc @mrclksr

> So it looks like the Hardware Probe finally got uploaded?

They always got uploaded, but there was a stupid bug crashing the GUI before it was showing the Probe URL. FIxed.
